modules/db/src/main/scala/gem/dao/package.scala

```diff
@@ -94,8 +97,12 @@ package object dao extends MoreTupleOps {
     AngleMetaAsSignedArcseconds.xmap(Offset.Q(_), _.toAngle)
 
   // Wavelength maps to an integer in angstroms
-  implicit val WavelengthMeta: Meta[Wavelength] =
-    Meta[Int].xmap(Wavelength.unsafeFromAngstroms, _.toAngstroms)
+  implicit val WavelengthMeta: Meta[WavelengthInÅngström] =
+    Meta[Int].xmap(i => Wavelength.fromAngstroms(UInt(i)), _.value.toInt)
+
+  // Wanted to do a generic quantity Meta but there is a bug with missing TypeTag
```

Comment:
I wanted to write a generic `Meta` but this raises some issues with missing `TypeTag`. @tpolecat indicates it is an issue with `shapeless` `HList`